app/scripts/modules/amazon/src/loadBalancer/details/targetGroupDetails.controller.ts

```diff
@@ -0,0 +1,80 @@
+import { IPromise, IQService, IScope, module } from 'angular';
+import { StateService } from 'angular-ui-router';
+
+import { Application, ILoadBalancer } from '@spinnaker/core';
+
+import { IAmazonApplicationLoadBalancer, ITargetGroup } from 'amazon/domain/IAmazonLoadBalancer';
+
+export interface ITargetGroupFromStateParams {
+  accountId: string;
+  region: string;
+  name: string;
+  loadBalancerName: string;
+  vpcId: string;
+}
+
+export class AwsTargetGroupDetailsController {
+  private targetGroupFromParams: ITargetGroupFromStateParams;
+  public application: Application;
+  public state = { loading: true };
+  public targetGroup: ITargetGroup;
+
+  constructor(private $scope: IScope,
+              private $q: IQService,
+              private $state: StateService,
+              targetGroup: ITargetGroupFromStateParams,
+              private app: Application) {
+    'ngInject';
+    this.application = app;
+    this.targetGroupFromParams = targetGroup;
+
+    this.app.ready().then(() => this.extractTargetGroup()).then(() => {
+      // If the user navigates away from the view before the initial extractTargetGroup call completes,
+      // do not bother subscribing to the refresh
+      if (!$scope.$$destroyed) {
+        app.getDataSource('loadBalancers').onRefresh($scope, () => this.extractTargetGroup());
+      }
+    });
+  }
+
+  public autoClose(): void {
+    if (this.$scope.$$destroyed) {
+      return;
+    }
+    this.$state.params.allowModalToStayOpen = true;
+    this.$state.go('^', null, {location: 'replace'});
+  }
+
+  public extractTargetGroup(): IPromise<void> {
```

Comment:
I copied extractLoadBalancer() and when I removed the Promise bits, forgot to remove the old return type too.